src/jdk.httpserver/share/classes/sun/net/httpserver/Request.java

```diff
@@ -50,11 +50,8 @@ class Request {
         os = rawout;
         do {
             startLine = readLine();
-            if (startLine == null) {
-                return;
-            }
             /* skip blank lines */
-        } while (startLine == null ? false : startLine.equals (""));
+        } while (startLine != null && startLine.equals (""));
```

Comment:
Drive-by comment;  Swapping the arguments for String.equals will handle the case of startLine == null.
`} while ("".equals(startLine));`

Reply:
Right - which reminds me `String.equals("")` is an anti-pattern that should probably be replaced by `String.isEmpty()`

Reply:
One or the other :-)

Reply:
LGTM.

We need a Strings utility class! ;-) Strings.isEmptyOrNull !

Reply:
`java.util.Objects.equals("", startLine)` would work as well. It covers any type.

Reply:
@RogerRiggs Thanks for your advice :smiley: